fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -578,78 +686,23 @@ private void clearBackendTransactions(TransactionState transactionState) {
-            for (TransactionState transactionState : allCommittedTransactionState) {
-                long dbId = transactionState.getDbId();
-                Database db = catalog.getDb(dbId);
-                if (null == db) {
-                    transactionState.setTransactionStatus(TransactionStatus.ABORTED);
-                    unprotectUpsertTransactionState(transactionState);
-                    continue;
-                }
-            }
         } finally {
             writeUnlock();
         }
-        
-        for (TransactionState transactionState : allCommittedTransactionState) {
-            boolean meetPublishPredicate = true;
-            long dbId = transactionState.getDbId();
-            Database db = catalog.getDb(dbId);
-            if (null == db) {
-                continue;
-            }
-            db.readLock();
-            try {
-                readLock();
-                try {
-                    for (TableCommitInfo tableCommitInfo : transactionState.getIdToTableCommitInfos().values()) {
-                        OlapTable table = (OlapTable) db.getTable(tableCommitInfo.getTableId());
-                        if (null == table) {
-                            LOG.warn("table {} is dropped after commit, ignore this table",
-                                     tableCommitInfo.getTableId());
-                            continue;
-                        }
-                        for (PartitionCommitInfo partitionCommitInfo : tableCommitInfo.getIdToPartitionCommitInfo().values()) {
-                            Partition partition = table.getPartition(partitionCommitInfo.getPartitionId());
-                            if (null == partition) {
-                                LOG.warn("partition {} is dropped after commit, ignore this partition",
-                                         partitionCommitInfo.getPartitionId());
-                                continue;
-                            }
-                            if (partitionCommitInfo.getVersion() != partition.getVisibleVersion() + 1) {
-                                meetPublishPredicate = false;
-                                break;
-                            }
-                        }
-                        if (!meetPublishPredicate) {
-                            break;
-                        }
-                    }
-                    if (meetPublishPredicate) {
-                        LOG.debug("transaction [{}] is ready to publish", transactionState);
-                        readyPublishTransactionState.add(transactionState);
-                    }
-                } finally {
-                    readUnlock();
-                }
-            } finally {
-                db.readUnlock();
-            }
-        }
-        return readyPublishTransactionState;
+
+        LOG.debug("there are {} transactions is COMMITTED", allCommittedTransactionState.size());
+
+        return allCommittedTransactionState.parallelStream()
```

Comment:
parallelStream() has a bad performance when num of elements in collections is small.
Don't you think `stream()` is enough?

Reply:
change back to `stream()`